jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -305,43 +305,38 @@ def get_link_type(self, data_label):
 
         return link_type
 
-    def _get_fov(self, data):
+    def _get_fov(self):
+        data = self.state.reference_data
+        if self.jdaviz_app._link_type != "wcs" or data.coords is None:
+            return
+
         # compute the mean of the height and width of the
         # viewer's FOV on ``data`` in world units:
         x_corners = [
             self.state.x_min,
             self.state.x_max,
-            self.state.x_min,
-            self.state.x_max
+            self.state.x_min
         ]
         y_corners = [
             self.state.y_min,
             self.state.y_min,
-            self.state.y_max,
             self.state.y_max
         ]
 
-        y_corners, x_corners = self._get_real_xy(
```

Comment:
BTW I thought this was clever but unfortunately, `_get_real_xy` was never meant to be vectorized. I dunno why but passing in scalar x/y in different output than passing in lists. Passing in scalars is what we have been using in main and the results are correct.